File: Java/src/jakopec/model/Partija.java
package jakopec.model;

import java.util.ArrayList;
import java.util.List;

public abstract class Partija extends Entitet {

    private int doKolikoSeIgra;
    private Lokacija lokacija;
    private Igrac unosi;
    private List<Mjesanje> mjesanja = new ArrayList<>();
    private List<Igrac> igraci = new ArrayList<>();


    public int getDoKolikoSeIgra() {
        return doKolikoSeIgra;
    }

    public void setDoKolikoSeIgra(int doKolikoSeIgra) {
        this.doKolikoSeIgra = doKolikoSeIgra;
    }

    public Lokacija getLokacija() {
        return lokacija;
    }

    public void setLokacija(Lokacija lokacija) {
        this.lokacija = lokacija;
    }

    public Igrac getUnosi() {
        return unosi;
    }

    public void setUnosi(Igrac unosi) {
        this.unosi = unosi;
    }

    public List<Mjesanje> getMjesanja() {
        return mjesanja;
    }

    public void setMjesanja(List<Mjesanje> mjesanja) {
        this.mjesanja = mjesanja;
    }

    public List<Igrac> getIgraci() {
        return igraci;
    }

    public void setIgraci(List<Igrac> igraci) {
        this.igraci = igraci;
    }

    public Rezulat getRezultat() {
        Rezulat rezulat = new Rezulat(getMjesanja().stream().mapToInt(x->x.getRezultat().getPrvi()).sum(),
                getMjesanja().stream().mapToInt(x->x.getRezultat().getDrugi()).sum());
        return rezulat;
    }

    public boolean isIgraGotova() {

        Rezulat rezultat = getRezultat();

        if(rezultat.isPocetak()){
            return false;
        }

        if(rezultat.getTreci()==0){
            return rezultat.getPrvi()==rezultat.getDrugi() ? false : rezultat.getPrvi()>getDoKolikoSeIgra() || rezultat.getDrugi()>getDoKolikoSeIgra();
        }else{
            if(rezultat.getPrvi()==rezultat.getDrugi() || rezultat.getPrvi()==rezultat.getTreci() || rezultat.getDrugi()==rezultat.getTreci()){
                return false;
            }

            if (rezultat.getPrvi()>getDoKolikoSeIgra() || rezultat.getDrugi()>getDoKolikoSeIgra() || rezultat.getTreci()>getDoKolikoSeIgra()){
                return true;
            }

        }

        return false;
    }
}

File: PHP/src/Jakopec/Partija.php
<?php


namespace Jakopec;


abstract class Partija extends Entitet
{
    private $doKolikoSeIgra;
    private $lokacija;
    private $unosi;
    private $mjesanja;
    private $igraci;

    /**
     * Partija constructor.
     * @param int $id
     * @param int $doKolikoSeIgra
     * @param Lokacija $lokacija
     * @param Igrac $unosi
     * @param array $mjesanja
     * @param array $igraci
     */
    public function __construct(int $id = 0, int $doKolikoSeIgra = 501, Lokacija $lokacija = null, Igrac $unosi = null, array $mjesanja = [], array $igraci = [])
    {
        $this->doKolikoSeIgra = $doKolikoSeIgra;
        $this->lokacija = $lokacija;
        $this->unosi = $unosi;
        $this->mjesanja = $mjesanja;
        $this->igraci = $igraci;
    }

    /**
     * @return Lokacija
     */
    public function getLokacija(): Lokacija
    {
        return $this->lokacija;
    }

    /**
     * @param Lokacija $lokacija
     */
    public function setLokacija(Lokacija $lokacija)
    {
        $this->lokacija = $lokacija;
    }

    /**
     * @return Igrac
     */
    public function getUnosi(): Igrac
    {
        return $this->unosi;
    }

    /**
     * @param Igrac $unosi
     */
    public function setUnosi(Igrac $unosi)
    {
        $this->unosi = $unosi;
    }

    /**
     * @return array
     */
    public function getIgraci(): array
    {
        return $this->igraci;
    }

    /**
     * @param array $igraci
     */
    public function setIgraci(array $igraci)
    {
        $this->igraci = $igraci;
    }

    public function isIgraGotova(): bool
    {

        $rezulat = $this->getRezultat();

        if ($rezulat->isPocetak()) {
            return false;
        }

        if ($rezulat->getTreci() == 0) {
            return $rezulat->getPrvi() == $rezulat->getDrugi() ?
                false : $rezulat->getPrvi() > $this->getDoKolikoSeIgra()
                || $rezulat->getDrugi() > $this->getDoKolikoSeIgra();
        } else {
            if ($rezulat->getPrvi() == $rezulat->getDrugi() ||
                $rezulat->getPrvi() == $rezulat->getTreci() ||
                $rezulat->getDrugi() == $rezulat->getTreci()) {
                return false;
            }

            if ($rezulat->getPrvi() > $this->getDoKolikoSeIgra() ||
                $rezulat->getDrugi() > $this->getDoKolikoSeIgra() ||
                $rezulat->getTreci() > $this->getDoKolikoSeIgra()) {
                return true;
            }

        }

        return false;
    }

    /**
     * @return int
     */
    public function getDoKolikoSeIgra(): int
    {
        return $this->doKolikoSeIgra;
    }

    /**
     * @param int $doKolikoSeIgra
     */
    public function setDoKolikoSeIgra(int $doKolikoSeIgra)
    {
        $this->doKolikoSeIgra = $doKolikoSeIgra;
    }

    public function getRezultat(): Rezultat
    {
        $rezultat = new Rezultat();
        $b = 0;
        foreach ($this->getMjesanja() as $mjesanje) {
            $b += $mjesanje->getRezultat()->getPrvi();
        }
        $rezultat->setPrvi($b);

        $b = 0;
        foreach ($this->getMjesanja() as $mjesanje) {
            $b += $mjesanje->getRezultat()->getDrugi();
        }
        $rezultat->setDrugi($b);

        return $rezultat;
    }

    /**
     * @return array
     */
    public function getMjesanja(): array
    {
        return $this->mjesanja;
    }

    /**
     * @param array $mjesanja
     */
    public function setMjesanja(array $mjesanja)
    {
        $this->mjesanja = $mjesanja;
    }

}